flow/layers/picture_layer.cc

```diff
@@ -17,6 +18,80 @@ PictureLayer::PictureLayer(const SkPoint& offset,
       is_complex_(is_complex),
       will_change_(will_change) {}
 
+void PictureLayer::Diff(DiffContext* context, const Layer* old_layer) {
+  DiffContext::AutoSubtreeRestore subtree(context);
+  if (!context->IsSubtreeDirty()) {
+    assert(old_layer);
+    auto prev = old_layer->as_picture_layer();
+    if (!Compare(context, this, prev)) {
+      context->MarkSubtreeDirty(prev->paint_region());
+    }
+  }
+  context->PushTransform(SkMatrix::Translate(offset_.x(), offset_.y()));
+  context->AddPaintRegion(picture()->cullRect());
+  set_paint_region(context->CurrentSubtreeRegion());
+}
+
+bool PictureLayer::Compare(DiffContext* context,
+                           const PictureLayer* l1,
+                           const PictureLayer* l2) {
+  const auto& pic1 = l1->picture_.get();
+  const auto& pic2 = l2->picture_.get();
+  if (pic1.get() == pic2.get()) {
+    context->statistics().AddSameInstancePicture();
+    return true;
+  }
+  auto op_cnt_1 = pic1->approximateOpCount();
+  auto op_cnt_2 = pic2->approximateOpCount();
+  if (op_cnt_1 != op_cnt_2 || pic1->cullRect() != pic2->cullRect()) {
+    context->statistics().AddNewPicture();
+    return false;
+  }
+
+  if (op_cnt_1 > 10) {
+    context->statistics().AddPictureTooComplexToCompare();
+    return false;
```

Comment:
I was worried that 10 may be too small a number, and this statistic will be useful for testing that theory with real world examples.